shell.go

```diff
@@ -62,14 +64,28 @@ func NewLocalShell() *Shell {
 }
 
 func NewShell(url string) *Shell {
-	c := &gohttp.Client{
-		Transport: &gohttp.Transport{
-			Proxy:             gohttp.ProxyFromEnvironment,
-			DisableKeepAlives: true,
-		},
+	var client *http.Client
+
+	tpt := &http.Transport{
+		Proxy:             http.ProxyFromEnvironment,
+		DisableKeepAlives: true,
+	}
+
+	maddr, err := ma.NewMultiaddr(url)
+	if err != nil {
+		client = &http.Client{Transport: tpt}
+		return NewShellWithClient(url, client)
+	}
+
+	if value, err := maddr.ValueForProtocol(ma.P_UNIX); err == nil {
```

Comment:
This can parse addrs like `/ip4/127.0.0.1/unix/...` as well. To be safe, let's call call `ma.SplitFirst(maddr)` and check to make sure it returns a _single_ component. Then check to make sure the component's protocol is P_UNIX.